testing/src/main/java/org/oppia/android/testing/TextInputAction.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.android.testing
+
+import android.content.Context
+import android.view.View
+import androidx.annotation.StringRes
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import javax.inject.Inject
+
+/**
+ * Action for matching error text and performing other actions on TextInputLayout in a test infrastructure-specific way.
+ */
+class TextInputAction @Inject constructor(val context: Context) {
+  // TODO(#1720): Move this to a companion object & use a test-only singleton injector to retrieve.
+
+  /**
+   * @param expectedErrorTextId is the string id which needs to match.
+   *
+   * @returns a [Matcher] that matches the specified string [expectedErrorTextId] with the view's error text.
```

Comment:
KDocs should always begin with a summary fragment before listing parameters/return value. However, given the simple contract we can actually fold both into just a summary fragment, e.g.:

```kotlin
  /**
   * Returns a [Matcher] that matches the string corresponding to the specified
   * [expectedErrorTextResId] with the view's error text.
   */
```

Ditto for the KDoc of the other method below.